waitgroup_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2011 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
```

Comment:
This seems odd 😁 

Shouldn't we add a header mentioning that this is based off the `sync` standard library tests ?

Reply:
I'm very uncomfortable with this licensing stuff and yes, I like your proposition.